xbmc/interfaces/Builtins.cpp

```diff
@@ -300,7 +300,8 @@ void CBuiltins::GetHelp(std::string &help)
 bool CBuiltins::ActivateWindow(int iWindowID, const std::vector<std::string>& params /* = {} */, bool swappingWindows /* = false */)
 {
   // don't activate a window if there are active modal dialogs
-  if (g_windowManager.HasModalDialog() && !g_windowManager.GetWindow(iWindowID)->IsDialog())
+  if (g_windowManager.HasModalDialog() &&
+      !(g_windowManager.GetWindow(iWindowID) != NULL && g_windowManager.GetWindow(iWindowID)->IsDialog()))
```

Comment:
this is basic behaviour of a modal dialog and needs to be done anyway in ActiveWindow.